app/curses_util.py

```diff
@@ -68,7 +68,9 @@
 KEY_HOME = curses.KEY_HOME
 KEY_END = curses.KEY_END
 KEY_PAGE_DOWN = curses.KEY_NPAGE
+KEY_SHIFT_PAGE_DOWN = curses.KEY_SNEXT
 KEY_PAGE_UP = curses.KEY_PPAGE
+KEY_SHIFT_PAGE_UP = curses.KEY_SPREVIOUS
```

Comment:
On Mac (iterm2 and Terminal) and ubuntu (whatever the default terminal in cinnamon is) the shift + page up/down is caught by the terminal software to do page up/down in the terminal history :( I get either no key input or KEY_NPAGE (for shift + page down; and page down; i.e. the same key code is sent).

Reply:
Your changes/code are correct. The bug is elsewhere and my code was just trying to compensate for bugs elsewhere.